python/ray/train/trainer.py

```diff
@@ -124,7 +121,8 @@ def __init__(
         self.logdir = self.create_logdir(logdir)
 
         # Setup executor.
-        backend_config = self._get_backend_config(backend)
+        self._backend_config = self._get_backend_config(backend)
+        self._backend = self._backend_config.backend_cls()
```

Comment:
Let's rename the `Backend` class in a future PR to make this less confusing!

Reply:
Yep makes sense!